File: kotlin/isogram/src/main/kotlin/Isogram.kt
object Isogram {

    val alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    fun isIsogram(input: String): Boolean {
        val alphabetMap = LinkedHashMap<Char, Int>();

        for (i in input) {
            val u = i.toUpperCase()

            if (alphabet.contains(u)) {
                alphabetMap.putIfAbsent(u, 0);
            }

            if (alphabetMap[u] != null) {
                alphabetMap[u] = alphabetMap[u]!! + 1;
            }
        }

        return alphabetMap.filter{ it.value > 1 }.size == 0
    }
}

File: go/isogram/isogram.go
// Package isogram has a function IsIsogram
package isogram

import "strings"

const ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

// IsIsogram takes an input string and returns a bool telling whether its an isogram or not
func IsIsogram(input string) bool {
	alphabetMap := make(map[rune]int, 26)

	input = strings.ToUpper(input)

	for _, i := range input {
		if strings.ContainsRune(ALPHABET, i) {
			alphabetMap[i] += 1
		}
	}

	for _, value := range alphabetMap {
		if value > 1 {
			return false
		}
	}

	return true
}
